Objects/typeobject.c

```diff
@@ -8059,6 +8060,53 @@ wrap_descr_delete(PyObject *self, PyObject *args, void *wrapped)
     Py_RETURN_NONE;
 }
 
+static PyObject *
+wrap_buffer(PyObject *self, PyObject *args, void *wrapped)
+{
+    PyObject *arg = NULL;
+
+    if (!PyArg_UnpackTuple(args, "", 1, 1, &arg)) {
+        return NULL;
+    }
+    Py_ssize_t flags = PyNumber_AsSsize_t(arg, PyExc_OverflowError);
```

Comment:
You need to manually check for integer overflow for int here, `Py_SAFE_DOWNCAST` actually does nothing in release mode, its just C level cast.  You need to manually check for `INT_MAX` before casting.